gtsfm/scene_optimizer.py

```diff
@@ -103,10 +102,16 @@ def create_computation_graph(
         image_graph: List[Delayed],
         camera_intrinsics_graph: List[Delayed],
         image_shape_graph: List[Delayed],
-        gt_cameras_graph: Optional[List[Delayed]] = None,
-        gt_scene_mesh: Optional[Trimesh] = None,
+        gt_gtsfm_data: Optional[GtsfmData] = None,
     ) -> Delayed:
         """The SceneOptimizer plate calls the FeatureExtractor and TwoViewEstimator plates several times."""
+        # Build cameras graph from GT GtsfmData.
+        # TODO (): remove later; this is just to conform to the required input of other functions.
+        gt_cameras_graph = (
+            [dask.delayed(gt_gtsfm_data.get_camera)(i) for i in gt_gtsfm_data._cameras.keys()]
```

Comment:
Maybe I should ask @ayushbaid , but what is our "standard representation" for cameras? Is it dicts or lists? why are converting one to another here?

Reply:
GtsfmData uses dicts to account for noncontiguous cameras ids, as some frames may be rejected during the verification process.

I convert the to lists here because the rest of the code expects it as a list, but I do agree that it's dangerous to have the same data as different data types.

Reply:
Yeah, so maybe we should update the rest of the code

Reply:
I spent a while trying to update the code so that we do not rely on converting the dictionary of cameras in `GtsfmData` to a List. However, almost all of the metric computation functions rely on Lists of Pose derived from the camera dictionary of GtsfmData.

I'm assuming this is why the `_number_images` attribute of the `GtsfmData` class was added in the first place: so that the cameras data could be converted to a list such that its the same length as the original ground truth data. I think this is should be saved for another PR.

Reply:
@ayushbaid

Reply:
Similarly, I think `_tracks` should be a dictionary as well.